core/src/main/java/hudson/model/Failure.java

```diff
@@ -54,6 +55,13 @@ public Failure(String message, boolean pre) {
         this.pre = pre;
     }
 
+    public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node, @Nullable Throwable throwable) throws IOException, ServletException {
```

Comment:
AFAIR Nullable is not needed at all as any object can be nulled by default. Maybe `Optional<Throwable>` at all?